### PR TITLE
feat(terraform): add BV, BP, BPA resources to sample configurations

### PR DESCRIPTION
**Pull request by niharika-98**
## Description

Fixes # b/378428041
Documention for backupdr resources:
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/backup_dr_backup_plan
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/backup_dr_backup_plan_association
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/backup_dr_backup_vault

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [x] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [x] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [x] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [x] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [x] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [x] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [x] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Comment by niharika-98** [comment]
@bharathkkb  @averbuks @tpdownes Can any of you please review this PR.

File: .github/CODEOWNERS
Comment (apeabody) [inline .github/CODEOWNERS:19]: nit: stray space
```suggestion
/network_connectivity/  @terraform-google-modules/dee-infra @terraform-google-modules/terraform-samples-reviewers
```

File: .github/CODEOWNERS
Comment (apeabody) [inline .github/CODEOWNERS:25]: nit: missing line return
```suggestion
/backupdr/              @terraform-google-modules/terraform-samples-reviewers

```

File: gcbdr/backupplan/main.tf
Comment (apeabody) [inline gcbdr/backupplan/main.tf:61]: nit
```suggestion
# [END backupdr_create_backupplan]

```

File: gcbdr/backupplanassociation/main.tf
Comment (apeabody) [inline gcbdr/backupplanassociation/main.tf:108]: nit
```suggestion
# [END backupdr_create_backupplanassociation]

```

File: gcbdr/backupvault/main.tf
Comment (apeabody) [inline gcbdr/backupvault/main.tf:37]: nit
```suggestion
# [END backupdr_create_backupvault]

```

**Comment by apeabody** [comment]
Thanks for the contribution @niharika-98!

I triggered the lint and CI tests.  For the lint, in addition to the nits I already identified, you probably need to run `terraform fmt` on the relevant files.

**Review by apeabody — APPROVED** [review]
`/test/`: LGTM (CODEOWNER)

@niharika-98 - You will still need approval from someone in @terraform-google-modules/terraform-samples-reviewers

File: .github/CODEOWNERS
Comment (apeabody) [inline .github/CODEOWNERS:25]: FYI: Per GH this team (`@terraform-google-modules/gcbdr-samples-team`) doesn't (currently) exist.

Comment (glasnt) [inline .github/CODEOWNERS:25]: The folder `backupdr` doesn't match the code you added `gcbdr`. Please rename the added code to `backupdr` (to also match the product code that's used in your region tags)

File: gcbdr/backupplan/main.tf
Comment (glasnt) [inline gcbdr/backupplan/main.tf:1]: For ease of reading, please rename each of these examples with underscore separated directory names (`backup_plan/main.tf`, `backup_plan_association/main.tf`, etc. )

File: gcbdr/backupplanassociation/main.tf
Comment (glasnt) [inline gcbdr/backupplanassociation/main.tf:19]: For this and all other resource names, these should reflect what a user might call them, as these will be displayed to the user in documentation, and should reflect what this type of resource would be called. 

For example, the instance might be `my-backedup-instance`, the vault might be `my-vault`.

File: gcbdr/backupvault/main.tf
Comment (glasnt) [inline gcbdr/backupvault/main.tf:23]: The description here is incorrect in the scope of this sample (there is only one vault in this file). 
```suggestion
  description                                = "This vault is provisioned by Terraform."
```

File: gcbdr/backupplan/main.tf
Comment (glasnt) [inline gcbdr/backupplan/main.tf:44]: Instead of having comments like this, add a region tag around the earlier `google_backup_dr_backup_vault` definition, and include that in your documentation before the backup_plan. This also allows your users to see both in context. (Additionally, since your plan references your vault, this resource cannot be provisioned on it's own without the associated dependency, so both should be wrapped in region tags and displayed on your documentation page.

File: .github/CODEOWNERS
Comment (niharika-98) [reply .github/CODEOWNERS:25]: Not sure why github is throwing error, but team is present
https://opensource.corp.google.com/github/team?org=terraform-google-modules&team=2184949506716467201

Comment (bharathkkb) [reply .github/CODEOWNERS:25]: It was an access issue, I have granted access to that team for this repo now

File: gcbdr/backupplan/main.tf
Comment (niharika-98) [reply gcbdr/backupplan/main.tf]: Update PR to reflect this

File: .github/CODEOWNERS
Comment (niharika-98) [reply .github/CODEOWNERS]: Updating to gcbdr everywhere, as it would be more appropriate

Comment (bharathkkb) [reply .github/CODEOWNERS]: > I would also suggest you match your team name with the product name

@glasnt this was already provisioned and the team mdb group was added. If not a hard requirement, would prefer to keep `gcbdr-samples-team` to reduce churn.

File: gcbdr/backupplan/main.tf
Comment (niharika-98) [reply gcbdr/backupplan/main.tf]: We have separate pagefor Backupvault. And user flow of documentation starts with BV and takes to BP. We mention BV terraform code in BV page, mentioning in BP page as well, looks repetitive?

Comment (niharika-98) [reply gcbdr/backupplan/main.tf]: Also, currently datasource support is not present for backupvault, we will ensure that, we update code to use datasource instead of resource creation multiple times in dependant resources

File: backupdr/backup_vault/main.tf
Comment (glasnt) [inline backupdr/backup_vault/main.tf:28]: For linting, either all `=` need to line up, or you need a new line space. 

```suggestion
  backup_minimum_enforced_retention_duration = "100000s"
  
  force_update  = "true"
  force_delete  = "true"
  allow_missing = "true"
}
```

**Review by glasnt — APPROVED** [review]

**Comment by glasnt** [comment]
All tests specifically added to this PR work, but due to adding content to /tests/, all tests are being run. Checking earlier CI runs, the new samples do work, so I will merge this PR, since all other tests have failed due to unrelated samples.

**Comment by apeabody** [comment]
> All tests specifically added to this PR work, but due to adding content to /tests/, all tests are being run. Checking earlier CI runs, the new samples do work, so I will merge this PR, since all other tests have failed due to unrelated samples.

@glasnt I think the error is from one of the new samples?
```
Error: "account_id" ("my-sa") must be between 6 and 30 characters long

  with google_service_account.default,
  on main.tf line 19, in resource "google_service_account" "default":
  19:   account_id   = "my-sa"
}
    validate.go:11: 
        	Error Trace:	/builder/home/go/pkg/mod/github.com/gruntwork-io/terratest@v0.47.2/modules/terraform/validate.go:11
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.3/pkg/tft/terraform.go:545
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.3/pkg/tft/terraform.go:585
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.3/pkg/tft/terraform.go:666
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.3/pkg/utils/stages.go:31
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.16.3/pkg/tft/terraform.go:666
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; 
        	            	Error: "account_id" ("my-sa") must be between 6 and 30 characters long
        	            	
        	            	  with google_service_account.default,
        	            	  on main.tf line 19, in resource "google_service_account" "default":
        	            	  19:   account_id   = "my-sa"
        	            	}
        	Test:       	TestSamples/3/backup_plan_association
```

**Comment by glasnt** [comment]
Ah, that was a later change. 

I'll create a fix in a new PR, which will allow us to also test these resources in isolation. 

For future new API PRs, might be an idea to do that PR first before the sample, for isolation purposes.

**Pull request merged**
